src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -59,6 +67,62 @@ internal void InitializeLicense(string licenseText)
             license = foundLicense;
         }
 
+        static bool FirstTimeUser()
+        {
+            // Check for the presence of HKCU:Software/NServiceBus
+            using (var regNsbRoot = Registry.CurrentUser.OpenSubKey(@"Software\NServiceBus"))
+            {
+                if (regNsbRoot == null)
+                {
+                    //Check for the presence of HKCU:SOFTWARE\ParticularSoftware
+                    using (var regParticularRoot = Registry.CurrentUser.OpenSubKey(@"Software\ParticularSoftware"))
+                    {
+                        if (regParticularRoot == null)
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Adding the async void to explicitly have a fire and forget model for invoking the web api.
+        /// </summary>
+        async void TrackFirstTimeUsageEvent()
+        {
+            // Get the current version of NServiceBus that's being used
+            var version = FileVersionRetriever.GetFileVersion(GetType());
+
+            // Report first time usage metric
+            Logger.InfoFormat("Reporting first time usage and version information to www.particular.net.  This call does not collect any personal information. For more details, see the License Agreement and the Privacy Policy available here: http://particular.net/licenseagreement.  This call will NOT be executed in production servers. It is executed only once for the first time, only when run in an interactive debugging mode.");
+            const string webApiUrl = "https://particular.net/api/ReportFirstTimeUsage";
+            var postData = $"version={version}";
+            try
+            {
+                var request = new HttpRequestMessage(HttpMethod.Post, webApiUrl)
+                {
+                    Content = new StringContent(postData, Encoding.UTF8, "application/x-www-form-urlencoded")
+                };
+
+                var response = await httpClient.SendAsync(request).ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
+            }
+            catch (Exception ex)
+            {
+                // For the end-user, this is not really an error that affects them. 
+                Logger.InfoFormat("Could not report first time usage statistics to www.particular.net: {0}", ex);
+            }
+        }
+
+        static void SetupFirstTimeRegistryKeys()
+        {
+            var regRoot = Registry.CurrentUser.CreateSubKey(@"Software\ParticularSoftware");
```

Comment:
A using statement does the exact same thing. If the value is null, then Dispose isn't called. Otherwise a using statement would throw a NRE when the value is null.